pkg/debug/start_debug.go

```diff
@@ -0,0 +1,162 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StartDebug(context *k8sutil.Context, clusterNamespace, deploymentName, alternateImageValue string) {
+	deployment, err := verifyDeploymentExists(context, clusterNamespace, deploymentName)
+	if err != nil {
+		fmt.Printf("Missing mon or osd deployment name %s. %v\n", deploymentName, err)
+		os.Exit(1)
+	}
+
+	if alternateImageValue != "" {
+		deployment.Spec.Template.Spec.Containers[0].Image = alternateImageValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deployment.Spec.Template.Spec.Containers[0].LivenessProbe = nil
+	deployment.Spec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if alternateImageValue != "" {
+		log.Printf("setting debug image to %s\n", alternateImageValue)
+		deployment.Spec.Template.Spec.Containers[0].Image = alternateImageValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deployment.Spec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deployment.Spec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(context, clusterNamespace, deployment); err != nil {
+		fmt.Printf("Failed to update deployment %s. %v\n", deployment.Name, err)
+		os.Exit(1)
+	}
+
+	deploymentPodName, err := podRunningWithLabel(context, clusterNamespace, deployment.Spec)
+	if deploymentPodName == "" || err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err := setDeploymentScale(context, clusterNamespace, deployment.Name, 0); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deploymentPodName)
+
+	err = waitForPodDeletion(context, clusterNamespace, deploymentPodName)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	debugDeploymentSpec := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deployment.Spec,
+	}
+
+	debugDeployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Create(ctx.TODO(), debugDeploymentSpec, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debugDeploymentSpec, err)
+		os.Exit(1)
+	}
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := setDeploymentScale(context, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
```

Comment:
How about logging at the end that it was successful? (perhaps the bash implementation does this?)

Reply:
this how it is in bash implementation